manager/integration/tests/test_ha.py

```diff
@@ -859,7 +860,137 @@ def test_single_replica_failed_during_engine_start():
+
+    volume = create_and_check_volume(client, volume_name, size=str(1 * Gi))
+    create_pv_for_volume(client, core_api, volume, pv_name)
+    create_pvc_for_volume(client, core_api, volume, pvc_name)
+
+    pod['metadata']['name'] = pod_name
+    pod['spec']['volumes'] = [{
+        'name': pod['spec']['containers'][0]['volumeMounts'][0]['name'],
+        'persistentVolumeClaim': {
+            'claimName': pvc_name,
+        },
+    }]
+
+    create_and_wait_pod(core_api, pod)
+    wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    data_path1 = "/data/file1"
+    write_pod_volume_random_data(core_api, pod_name,
+                                 data_path1, DATA_SIZE_IN_MB_1)
+    data_md5sum1 = get_pod_data_md5sum(core_api, pod_name, data_path1)
+    snap1 = volume.snapshotCreate()
+
+    data_path2 = "/data/file2"
+    write_pod_volume_random_data(core_api, pod_name,
+                                 data_path2, DATA_SIZE_IN_MB_1)
+    data_md5sum2 = get_pod_data_md5sum(core_api, pod_name, data_path2)
+    snap2 = volume.snapshotCreate()
+
+    data_path3 = "/data/file3"
+    write_pod_volume_random_data(core_api, pod_name,
+                                 data_path3, DATA_SIZE_IN_MB_1)
+    data_md5sum3 = get_pod_data_md5sum(core_api, pod_name, data_path3)
+    snap3 = volume.snapshotCreate()
+
+    volume = client.by_id_volume(volume_name)
+    host_id = get_self_host_id()
+
+    for replica in volume.replicas:
+        if replica.hostId == host_id:
+            break
+
+    replica_data_path = replica.dataPath
+    replica_name = replica.name
+    snap2_meta_file = replica_data_path + \
+        "/volume-snap-" + \
+        snap2.name + ".img.meta"
+
+    command = ["dd", "if=/dev/zero",
+               "of="+snap2_meta_file,
+               "count=" + str(DATA_SIZE_IN_MB_1),
+               "bs=1M"]
+    subprocess.check_call(command)
+
+    delete_and_wait_pod(core_api, pod_name)
+    wait_for_volume_detached(client, volume_name)
+
+    create_and_wait_pod(core_api, pod)
+    wait_for_volume_degraded(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+    for repl in volume.replicas:
+        if repl.name == replica_name:
+            break
+
+    assert repl.running is False
+    assert repl.failedAt != ''
+
+    wait_for_volume_healthy(client, volume_name)
+
+    res_data_md5sum1 = get_pod_data_md5sum(core_api, pod_name, data_path1)
+    assert data_md5sum1 == res_data_md5sum1
+
+    res_data_md5sum2 = get_pod_data_md5sum(core_api, pod_name, data_path2)
+    assert data_md5sum2 == res_data_md5sum2
+
+    res_data_md5sum3 = get_pod_data_md5sum(core_api, pod_name, data_path3)
+    assert data_md5sum3 == res_data_md5sum3
+
+    data_path4 = "/data/file4"
+    write_pod_volume_random_data(core_api, pod_name,
+                                 data_path4, DATA_SIZE_IN_MB_1)
+    data_md5sum1 = get_pod_data_md5sum(core_api, pod_name, data_path4)
```

Comment:
Should it be named as `data_md5sum4` rather than `data_md5sum1`? And it seems that this returned variable is not used in the test.

Reply:
fixed